docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+  </section>
+)
+
+export default WithContext
+```
+
+Start the development server with `gatsby develop` and visit `localhost:8000/page-with-context/` to see the created page:
+
+![Screenshot of a page with hard-coded context.](./images/why-gql-with-context.png)
+
+In some cases, this approach may be enough. However, it’s often necessary to add create pages from data that’s not hard-coded.
+
+## Create pages from JSON with images
+
+In many cases, the data for pages won’t be hard-coded into `gatsby-node.js`. More likely it will come from an external source, such as a JSON file or third-party API.
+
+For example, you might have a JSON file with post data:
+
+```json:title=data/products.json
+[
+  {
+    "title": "Vintage Purple Tee",
+    "slug": "vintage-purple-tee",
+    "description": "<p>Keep it simple with this vintage purple tee.</p>",
+    "price": "$10.00",
+    "image": "/images/amberley-romo-riggins.jpg"
+  },
+  {
+    "title": "Space Socks",
+    "slug": "space-socks",
+    "description": "<p>Get your feet into these spaced-out black socks with a Gatsby purple border and heel.</p>",
+    "price": "$10.00",
+    "image": "/images/erin-fox-and-sullivan.jpg"
+  },
+  {
+    "title": "This Purple Hat Is Blazing Fast",
+    "slug": "purple-hat",
+    "description": "<p>Add more blazingly blazing speed to your wardrobe with this solid purple laundered chino twill hat.</p>",
+    "price": "$10.00",
+    "image": "/images/david-bailey-cat-hat.jpg"
+  }
+]
+```
+
+The images need to be added to the `/static/images/` folder. (This is where things start to get hard to manage — the JSON and the images aren’t in the same place.)
+
+Once the JSON and the images are added, you can create product pages by importing the JSON into `gatsby-node.js` and loop through the entries to create pages:
+
+```js:title=gatsby-node.js
+exports.createPages = ({ actions: { createPage } }) => {
+  const products = require("./data/products.json")
+  products.forEach(product => {
+    createPage({
+      path: `/product/${product.slug}/`,
+      component: require.resolve("./src/templates/product.js"),
+      context: {
+        title: product.title,
+        description: product.description,
+        image: product.image,
+        price: product.price,
+      },
+    })
+  })
+}
+```
+
+The product template still uses `pageContext` to display the product data:
+
+```jsx:title=src/templates/product.js
+import React from "react"
+
+const Product = ({ pageContext }) => (
+  <div>
+    <h1>{pageContext.title}</h1>
+    <img
+      src={pageContext.image}
+      alt={pageContext.title}
+      style={{ float: "left", marginRight: "1rem", width: 150 }}
+    />
+    <p>{pageContext.price}</p>
+    <div dangerouslySetInnerHTML={{ __html: pageContext.description }} />
+  </div>
+)
+
+export default Product
+```
+
+Run `gatsby develop`, then open `localhost:8000/product/space-socks/` to see one of the generated products.
```

Comment:
As long as every starter supports this command, I'm on board with it. I'm going to do one big update for all instances (already ran the find-and-replace), so I'm not going to merge this suggestion, though.

Reply:
> As long as every starter supports this command, I'm on board with it

They do! They also support `npm start` which is maybe a bit simpler, too.

> I'm going to do one big update for all instances (already ran the find-and-replace), so I'm not going to merge this suggestion, though.

I actually already did this! Some instances were intentionally left because they're referring to the command itself rather than _using_ the command. See gatsbyjs/gatsby#11556

Reply:
> I actually already did this!

I meant in this article only — just pushed that update. 😄